src/hooks/useQuota/useQuota.test.tsx

```diff
@@ -0,0 +1,488 @@
+        {
+          category: "toilet-paper",
+          identifierInputs: [
+            {
+              label: "first",
+              value: "first identifier",
+              textInputType: "STRING",
+              scanButtonType: "BARCODE"
+            },
+            {
+              label: "last",
+              value: "last identifier",
+              textInputType: "STRING",
+              scanButtonType: "BARCODE"
+            }
+          ],
+          transactionTime,
+          quantity: 1
+        },
+        {
+          category: "chocolate",
+          identifierInputs: [],
+          transactionTime,
+          quantity: 15
+        }
+      ]);
+    });
+
+    it("should have quota state be NO_QUOTA when no quota is available", async () => {
+      expect.assertions(3);
+      mockGetQuota.mockReturnValueOnce(mockQuotaResSingleIdNoQuota);
+
+      const ids = ["ID1"];
+      const { result, waitForNextUpdate } = renderHook(
+        () => useQuota(ids, key, endpoint),
+        { wrapper }
+      );
+      expect(result.current.quotaState).toBe("FETCHING_QUOTA");
+
+      await waitForNextUpdate();
+      expect(result.current.quotaState).toBe("NO_QUOTA");
+      expect(result.current.quotaResponse?.remainingQuota).toStrictEqual([
+        {
+          category: "toilet-paper",
+          identifierInputs: [],
+          transactionTime,
+          quantity: 0
+        },
+        {
+          category: "chocolate",
+          identifierInputs: [],
+          transactionTime,
+          quantity: 0
+        }
+      ]);
+    });
+    it("should have quota state be NO_QUOTA when quota received is invalid", async () => {
+      expect.assertions(3);
+      mockGetQuota.mockReturnValueOnce(mockQuotaResSingleIdInvalidQuota);
+
+      const ids = ["ID1"];
+      const { result, waitForNextUpdate } = renderHook(
+        () => useQuota(ids, key, endpoint),
+        { wrapper }
+      );
+      expect(result.current.quotaState).toBe("FETCHING_QUOTA");
+
+      await waitForNextUpdate();
+      expect(result.current.quotaState).toBe("NO_QUOTA");
+      expect(result.current.quotaResponse?.remainingQuota).toStrictEqual([
+        {
+          category: "toilet-paper",
+          identifierInputs: [],
+          transactionTime,
+          quantity: -1
+        },
+        {
+          category: "chocolate",
+          identifierInputs: [],
+          transactionTime,
+          quantity: 15
+        }
+      ]);
+    });
+    it("should set quota state to be NOT_ELIGIBLE when NotEligibleError is thrown, and would not continue with fetching quota", async () => {
+      expect.assertions(1);
+
+      const ids = ["ID_NOT_ELIGIBLE"];
+
+      mockGetQuota.mockImplementation(() => {
+        mockIdNotEligible(ids[0]);
+      });
+
+      const { result } = renderHook(() => useQuota(ids, key, endpoint), {
+        wrapper
+      });
+
+      expect(result.current.quotaState).toBe("NOT_ELIGIBLE");
+    });
```

Comment:
Hmmm not sure what this test is doing, what does it mean by would not continue with fetching quota? HMW make this clearer?

Reply:
I have renamed this to: `"should set quota state to NOT_ELIGIBLE when NotEligibleError is thrown, and does not update quota response"`.

I have also added another test case to test that if an existing quota response exists and `NotEligibleError` is thrown, the quota state is `NOT_ELIGIBLE`, but the previous quota response is not overridden. Let me know if these tests are sound.